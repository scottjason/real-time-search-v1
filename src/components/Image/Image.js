import React from 'react';
import {Img} from './Image.style';
import LazyLoad from 'react-lazyload';

const Image = ({ item }) => {
  return(
    <LazyLoad height={300}>
      <Img imageUrl={item.imageUrl} />
    </LazyLoad>
  )
}

export default Image;
