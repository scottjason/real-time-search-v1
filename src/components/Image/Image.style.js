import styled from 'styled-components';
import media from 'styled-media-query';
const hdUrl = 'https://images.unsplash.com/photo-1478562853135-c3c9e3ef7905';

export const Img = styled.div`
  min-height: 300px;
  background-size: cover;
  background-repeat: no-repeat;
  background-postion: center;
  background-image: url(${props => props.imageUrl});
  border: 1px solid rgba(225, 225, 225, .4);
  transition: all ease 150ms;
  box-sizing: border-box;
  ${media.lessThan('medium')`
    /* screen width is less than 768px (medium) */
    min-height: 100px;
  `}
`;
