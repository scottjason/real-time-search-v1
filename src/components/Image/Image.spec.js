import React from 'react';
import { shallow, mount } from 'enzyme';
import Image from './Image';

/**
 *
 * Unit Testing <Image/>
 *
*/

describe('Image', () => {

  let wrapper;
  let mockImages = [
    {id: 0, imageUrl: 'https://picsum.photos/500'},
  ];

  beforeAll(() => {
    wrapper = mount(<Image item={mockImages[0]} />)
  });

  /**
  * High-Level Checks
  */
  test('it should exist', () => {
    expect(wrapper).toExist(true);
  });

  test('it should have correct name', () => {
    expect(wrapper.name()).toEqual('Image');
  });

  test('it should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  /**
  * Props
  */
  test('it should have item prop defined', () => {
    expect(wrapper.props().item).toBeDefined();
  });
  
  test('item should have imageUrl attr defined', () => {
    expect(wrapper.props().item.imageUrl).toBeDefined();
  });

});