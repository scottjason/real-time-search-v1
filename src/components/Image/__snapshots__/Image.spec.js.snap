// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image it should match snapshot 1`] = `
<Image
  item={
    Object {
      "id": 0,
      "imageUrl": "https://picsum.photos/500",
    }
  }
>
  <LazyLoad
    height={300}
    offset={0}
    once={false}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <div
      className="lazyload-placeholder"
      style={
        Object {
          "height": 300,
        }
      }
    />
  </LazyLoad>
</Image>
`;
